pkg/agents/team3/messages.go

```diff
@@ -1 +1,186 @@
 package team3
+
+import (
+	"math/rand"
+
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+	"github.com/SOMAS2021/SOMAS2021/pkg/messages"
+)
+
+//Upon receipt of message define affected emotions
+// ACK MESSAGE
+//If x time passed no message received/acked morale decrease
+//Include if ack message same user ID occurs x+1 times, morale increase
+//If stubborness = y+1, discard, a.k.a. leave unread
+func ticklyMessage(a *CustomAgent3) {
+	r := rand.Intn(5)
+	switch r {
+	case 0:
+		msg := messages.NewAskFoodTakenMessage(a.BaseAgent().ID(), a.Floor(), a.Floor()-1)
+		a.SendMessage(msg)
+		a.Log("I sent a message", infra.Fields{"message": "AskFoodTaken"})
+	case 1:
+		msg := messages.NewAskHPMessage(a.BaseAgent().ID(), a.Floor(), a.Floor()-1)
+		a.SendMessage(msg)
+		a.Log("I sent a message", infra.Fields{"message": "AskHP"})
+	case 2:
+		msg := messages.NewAskIntendedFoodIntakeMessage(a.BaseAgent().ID(), a.Floor(), a.Floor()-1)
+		a.SendMessage(msg)
+		a.Log("I sent a message", infra.Fields{"message": "AskIntendedFoodIntake"})
+	case 3:
+		msg := messages.NewRequestLeaveFoodMessage(a.BaseAgent().ID(), a.Floor(), a.Floor()+1, 10)
+		a.SendMessage(msg)
+		a.Log("I sent a message", infra.Fields{"message": "RequestLeaveFood"})
+	default:
+		msg := messages.NewRequestTakeFoodMessage(a.BaseAgent().ID(), a.Floor(), a.Floor()+1, 10) //make func to determine value
+		a.SendMessage(msg)
+		a.Log("I sent a message", infra.Fields{"message": "RequestTakeFood"})
+	}
+}
+func message(a *CustomAgent3) {
```

Comment:
```suggestion
func (a *CustomAgent3)message() {
```

Reply:
I feel like this is the way it should be done for all of the functions but this is just a suggestion.

Reply:
Agree

Reply:
Agree